tests/delete/conftest.py

```diff
@@ -149,10 +170,33 @@ def aws_rhcos_push_item(ami_release: AmiRelease, security_group: AmiSecurityGrou
 
 
 @pytest.fixture
-def pub_response(aws_rhcos_push_item: AmiPushItem, aws_push_item: AmiPushItem) -> List[AmiPushItem]:
+def vhd_push_item(release_params: Dict[str, Any], push_item_params: Dict[str, Any]) -> AmiPushItem:
+    """Return a minimal VHDPushItem."""
+    release = AmiRelease(**release_params)
+    push_item_params.update(
+        {
+            "name": "azure-testing.vhd.xz",
+            "release": release,
+            "src": "mnt/azure/azure-testing.vhd.xz",
+            "dest": ["azure-testing"],
+            "build": "azure-testing",
+        }
+    )
+    return AmiPushItem(**push_item_params)
```

Comment:
Why is this returning `AmiPushItem` instead of `VHDPushItem`?

Reply:
The pushsource code is set to return an AmiPushItem. Which has all the info we need to run this delete. It didn't seem atm to be useful to spend a lot of time updating that.

Reply:
Yes, but this is the `vhd_push_item` fixture, so it would return a `VHDPushItem`, no?

This way you're testing an `AmiPushItem` deletion for Azure which shouldn't work

Reply:
Ohhhhh, I got it now!!

Although I think that's a bit hackish, since ideally `pushsource` should be able to produce either `AmiPushItem` or `VHDPushItem` from `clouds.json`... Worst case, if that's too complicated at least a `VMIPushItem` which we could cast in lower environment, like this tooling, but preferable producing the correct typing from upper library...

@lslebodn WDYT? IMHO I believe we must patch `pushsource` first